node/bindnode/repr.go

```diff
@@ -589,10 +617,9 @@ func (w *_assemblerRepr) AssignString(s string) error {
 				unionSetMember(w.val, idx, valPtr)
 				return nil
 			}
-
-			return w2.AssignString(value)
+			return w2.AssignString(remainder)
 		}
-		panic("TODO: GetMember result is missing?")
+		return fmt.Errorf("data doesn't match union type: type %s couldn't match any of its prefixes to the data %q", w.schemaType.Name(), s)
```

Comment:
I'd also try to make this error shorter. you say "does not match", "data", and "type" twice, so the first sentence could be removed, for example